### PR TITLE
Historical data page

### PR DESCRIPTION
**Pull request by upatricck**
**Historical data Page 
**This is a page to display daily data average for the whole year

* You can view data for all the daily data columns for the whole year
* The page gives the total number of alerts, average power and average pv

To be done
* To make a day clickable so that you can get, more information about the day

### CODE REVIEW

File: seshdash/generateData.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
+from django.utils import timezone
```
Comment (jlark) [inline seshdash/generateData.py]: This file looks like a duplicate of the generate data file in /seshdash/tests/
Can you merge the files so that there is only one module that has all the data generation functions we need? I would put it under. /seshdash/tests/ as data generation is related to testing

File: seshdash/generateData.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+from seshdash.tests.data_generation import get_random_int, get_random_binary, get_random_interval, generate_date_array, get_random_float
+
+
+def create_test_data(site_id):
+        i = Influx()
+        start_date = timezone.now() - timedelta(weeks=55)
+        data_point_dates = generate_date_array(start=start_date, interval=24, units='hours')
+        voltage_in = 220
+        voltage_out = 220
+   	total_items = len(data_point_dates)
```
Comment (jlark) [inline seshdash/generateData.py]: the spacing seems to be incorrect for this file.

Can you check?

File: seshdash/generateData.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+from seshdash.data.db.influx import Influx
+from datetime import timedelta, datetime
+from django.forms import model_to_dict
+from seshdash.tasks import run_aggregate_on_historical
+from seshdash.models import Daily_Data_Point, BoM_Data_Point, Sesh_Site
+from seshdash.utils.time_utils import get_time_interval_array
+from seshdash.tests.data_generation import get_random_int, get_random_binary, get_random_interval, generate_date_array, get_random_float
+
+
+def create_test_data(site_id):
```
Comment (jlark) [inline seshdash/generateData.py]: Please be a bit more specific with the function name. This function is generating test data for what? Individual days or data points? The name should be descriptive to this

File: seshdash/models.py [hunk 4]
```diff
@@ -242,9 +242,21 @@ class Meta:
...
     """
 
+    UNITS_DICTIONARY = {
+        "daily_battery_charge": "",
+        "daily_grid_outage_n": "minute",
+        "daily_grid_outage_t": "",
+        "daily_grid_usage": "",
+        "daily_no_of_alerts": "alert",
+        "daily_power_cons_pv": "",
+        "daily_power_consumption_total": "",
```
Comment (jlark) [inline seshdash/models.py]: did you leave the units for these guys empty?

File: seshdash/sesh-ui/js/sesh_react.js [hunk 5]
```diff
@@ -55,8 +55,6 @@ var alertsJsonData = {csrfmiddlewaretoken: csrftoken,
                       site_id: active_site_id,
                      };
 
-getLatestAlerts();
```
Comment (jlark) [inline seshdash/sesh-ui/js/sesh_react.js]: Why was this removed? `getLatestBoMData` and  `getLatestAlerts`

File: seshdash/views.py [hunk 6]
```diff
@@ -615,3 +616,77 @@ def get_latest_bom_data(request):
...
+        return render(request, 'seshdash/historical-data.html', context_dict);
+
+
+def get_alerts_for_year(site):
+    """ Returns a json object containing the number of alerts for a year range """
+ 
+    # NOTE: When using datetime object date_range does not include the last date
+    now = datetime.datetime.now()
+    year_before = now - relativedelta(years=1)
+    number_of_alerts = Sesh_Alert.objects.filter(date__range=[
```
Comment (jlark) [inline seshdash/views.py]: you could use simply the `year` filter instead of a date range here and let django handle this for you.

File: seshdash/views.py [hunk 7]
```diff
@@ -615,3 +616,77 @@ def get_latest_bom_data(request):
...
+def get_avg_field_year(site, field):
+    """ Returns the average of a field for a year range in Daily Data Point"""
+    avg_field_yield = Daily_Data_Point.objects.filter(site=site)[:365].aggregate(Avg(field)).values()[0]
+    return avg_field_yield
+
+def get_historical_dict(column='daily_pv_yield'):
+    unit = Daily_Data_Point.UNITS_DICTIONARY[column]
+    print "The unit is: ",
+    print unit
+    sites = Sesh_Site.objects.all();
```
Comment (jlark) [inline seshdash/views.py]: Getting all sites is a problem. Imagine there are many users and some sites belong to them and some sites don't

**Comment by jlark** [comment]
Good work @upatricck , please checkout comments , fix testing and we should be good to go!

File: seshdash/sesh-ui/js/sesh_react.js (hunk 5 above)
Comment (upatricck) [inline seshdash/sesh-ui/js/sesh_react.js]: I removed them for testing and forgot to put them again.
Now there are back

File: templates/seshdash/historical-data.html [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+                <h2 class="historical-header"> Historical Data </h2>
+
+            {% endblock %}
+            </div>
+        </div>
+
+       <div class="filter-form">
+         <div class="data-sort-container">
+            <p class="sort-help-text"> Select an option to sort by </p>
+            <select class="form-control" id="data-sort">
```
Comment (jlark) [inline templates/seshdash/historical-data.html]: can you make this dynamic? These values are hard coded. I would rather they get sent back in the context dict and this is generated dynamically.

File: templates/seshdash/historical-data.html [hunk 9]
```diff
@@ -0,0 +1,96 @@
...
+                 {{ sort_dict }}
+                 {% for key, value in sort_dict.items %}	
+                      <option value="{{key}}">{{value}}</option>
+                 {% endfor %}
+                 <!--
+                 <option value="daily_battery_charge">Battery Charge</option>
```
Comment (jlark) [inline templates/seshdash/historical-data.html]: can you delete these comments?

File: templates/seshdash/historical-data.html [hunk 10]
```diff
@@ -0,0 +1,96 @@
...
+            </div>
+        </div>
+
+
+       <div class="chart-data-container">
+           
+           
+           <!--
+           <div class="panel panel-default ">
+               <div class="panel-heading"> <strong> Site One </strong> </div>
```
Comment (jlark) [inline templates/seshdash/historical-data.html]: same  here
